pkg/controller/v1beta1/inferenceservice/scheduler/shard_manager.go

```diff
@@ -0,0 +1,20 @@
+package scheduler
+
+import v1beta1api "github.com/kubeflow/kfserving/pkg/apis/serving/v1beta1"
+
+type ShardingStrategy string
+
+const (
+	Memory ShardingStrategy = "Memory"
+)
+
+type ShardManager struct {
+	Strategy ShardingStrategy
+}
+
+func (v *ShardManager) GetShardId(isvc *v1beta1api.InferenceService) []int {
```

Comment:
I think this function probably needs both a the TrainedModel and the inferenceservice.

Reply:
Yeah I created two functions 
```
// Return all the shard ids for an InferenceService
func (v *ShardManager) GetShardIdsForInferenceService(isvc *v1beta1api.InferenceService) []int {
	//TODO implement sharding logic for InferenceService
	//Currently each InfereceService only has one shard with id 0
	ids := []int{0}
	return ids
}

// Return a TrainedModel's shardId
func (v *ShardManager) GetShardIdForTrainedMode(trainedModel *v1beta1api.TrainedModel) int {
	//TODO to be implemented in another PR
	return 0
}
```

Reply:
InferenceService will contain a list of shards but one TrainedModel will be assigned to one shard.